pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go

```diff
@@ -297,6 +300,8 @@ func (r *ReconcileNetworkAddonsConfig) Reconcile(request reconcile.Request) (rec
 		return reconcile.Result{}, err
 	}
 
+	monitoring.TrackMonitoringComponents(&networkAddonsConfig.Spec, r.clusterInfo.MonitoringAvailable)
```

Comment:
The condition MonitoringAvailable should be checked here. If it is not available, we should not invoke the module at all.  That way we encapsulate the helper module and isolate it from the dependencies. Would that make sense?

Reply:
I don't mind, but note we do it differently on render functions (see [here](https://github.com/kubevirt/cluster-network-addons-operator/blob/main/pkg/network/kubemacpool.go#L62-L64))
Do you think we should change it there as well? (in a different PR of course)

Reply:
Done